src/app/modules/main/communities/minting/models/token_item.nim

```diff
@@ -0,0 +1,100 @@
+import strformat
+
+type
+  TokenType* {.pure.} = enum
+    #ERC20,
+    ERC721
+
+type
+  MintingState* {.pure.} = enum
+    InProgress,
+    Minted,
+    Failed
+
+type
+  TokenItem* = ref object
+    tokenType: TokenType
+    tokenAddress: string
+    name: string
+    description: string
+    icon: string
+    supply: string
```

Comment:
Could the supply be different than an int??

Reply:
Changed to int